### DISCUSSION TITLE
Feeback required for upcoming release 0.10.0

### DISCUSSION THREAD

**Discussion by konsoletyper** [konsoletyper/teavm#908] [Announcements]
Hi!

I'm glad to announce that release 0.10.0 is to come soon. I implemented all the features I planned and now I only need your feedback. So if it's possible, please, try latest developer preview (0.10.0-dev-12) and tell if you had some issues.

The most notable features of upcoming release are:

* Strings are represented by thin wrappers around JavaScript strings, which allows to improve performance in several operations and in interoperability between Java and JS
* JS backend was refactored to improve code generation, which allows to decrease generated code size
* JS backend produces ES6 code
* Gradle supports dev server (see new tasks `javaScriptDevServer` and `stopJavaScriptDevServer`
* Allow to generate JS modules ([see documentation](https://teavm.org/docs/runtime/js-modules.html))
* JSO classes can now be represented by non-abstract classes. In this case additional `@JSClass` annotation required on class. To represent member methods, `native` methods can be used instead of `abstract`. This allows to map JS to Java constructors and JS static methods. `@JSModule` annotation can be specified for class to indicate that class should be imported from given module, not taken from global context.
* Vararg methods are supported in JSO
* `instanceof` now works properly for JSO classes, as well as casts. This only works for classes, not for interfaces. To turn off `instanceof` for particular class, add `@JSClass(transparent = true)`
* `@JSPrimitiveType` can be used to map Java class to JS primitive (this affects `instanceof` and casts)
* `@JSTopLevel` annotation can be applied to static methods to indicate that they aren't mapped to JS static methods, but rather to global scope. `@JSTopLevel` can be combined with `@JSModule` to import functions from modules.
* Changes in built-in JSO declarations according to improvements in JSO

**Comment by SquidDev** [comment]
Just updated to the latest version, and it appears to be working perfectly for me. Thank you for your hard work, the support for ESM is so useful! :heart:
